src/test/java/org/gridsuite/securityanalysis/server/SecurityAnalysisControllerTest.java

```diff
@@ -93,15 +95,22 @@ public void setUp() {
 
         // action service mocking
         given(actionsService.getContingencyList(CONTINGENCY_LIST_NAME, NETWORK_UUID))
-                .willReturn(Mono.just(SecurityAnalysisFactoryMock.CONTINGENCIES));
+                .willReturn(Flux.fromIterable(SecurityAnalysisFactoryMock.CONTINGENCIES));
         given(actionsService.getContingencyList(CONTINGENCY_LIST2_NAME, NETWORK_UUID))
-                .willReturn(Mono.just(SecurityAnalysisFactoryMock.CONTINGENCIES));
+                .willReturn(Flux.fromIterable(SecurityAnalysisFactoryMock.CONTINGENCIES));
+        given(actionsService.getContingencyList(CONTINGENCY_LIST_ERROR_NAME, NETWORK_UUID))
+                .willReturn(Flux.error(new DataBufferLimitException("Exceeded limit on max bytes to buffer : 262144")));
 
         // UUID service mocking to always generate the same result UUID
         given(uuidGeneratorService.generate()).willReturn(RESULT_UUID);
 
         // mock the powsybl security analysis service
         configService.setSecurityAnalysisFactoryClass(SecurityAnalysisFactoryMock.class.getName());
+
+        // purge messages
+        while (output.receive(1000) != null) {
+            output.receive(1000);
```

Comment:
why pull out 2 messages at each loop iteration ? Maybe explain with a comment

Reply:
it is a mistake replaced by 
```
while (output.receive(1000) != null) {
    continue;
}
```